lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -31,6 +31,22 @@ def default_priority
     self.priority = MiqAeDomain.highest_priority(tenant) + 1 unless priority
   end
 
+  def version
+    system = ae_namespaces.detect { |ns| ns.name.downcase == 'system' }
+    about = system.try(:ae_classes).try(:detect) { |cls| cls.name.downcase == 'about' }
```

Comment:
These first two lines are the same as in the `about_file_name` method.  They could be extracted into a separate method that returns both items.

`_system, about = system_about_lookup`  (I'm sure there is a better name than this.)